### PR TITLE
Bug 1332711 Added more info to openID auth topic

### PR DESCRIPTION
**Pull request by bfallonf**
For: https://bugzilla.redhat.com/show_bug.cgi?id=1332711

I'm not sure if I'm off the mark here. @liggitt you were in the customer case. Could you please verify my changes here? There's a lot of talk about identity and identity provider objects in the case, and I don't want to mess them up.

Also, is the `claims` part of the example accurate? Should the `sub` (at minimum) parts have replaceables, or better examples? Let me know if I'm not being specific enough.

### CODE REVIEW

File: install_config/configuring_authentication.adoc [hunk 1]
```diff
@@ -1720,13 +1718,12 @@ must be allowed to redirect to `<master>/oauth2callback/<identityProviderName>`.
 <6> The client secret. This value may also be provided in an
 xref:../install_config/master_node_configuration.adoc#master-node-configuration-passwords-and-other-data[environment
 variable, external file, or encrypted file].
-<7> Use the value of the `sub` claim in the returned `id_token` as the user's
```
Comment (liggitt) [inline install_config/configuring_authentication.adoc]: the existing text is more accurate

**Comment by liggitt** [comment]
There are two examples, and the second goes into a lot of details and demonstrates the breadth of potential settings, but is not normative, which is why it wasn't made the primary example

File: install_config/configuring_authentication.adoc [hunk 2]
```diff
@@ -1665,8 +1665,8 @@ with a non-empty value is used. The
 link:http://openid.net/specs/openid-connect-core-1_0.html#StandardClaims[standard claims] are:
 
 [horizontal]
-`sub`:: The user identity.
-`preferred_username`:: The preferred user name when provisioning a user.
+`sub`:: The ID of the user within the configured identity provider. For example `user1`. This can be created before the authorization process. When added to the identity object, this is prefixed by the identity provider object. For example `myidp:user1`. If the identity object does not exist, so, if the user has not been created, then the prefix depends on the `preferred_username`.
```
Comment (liggitt) [inline install_config/configuring_authentication.adoc]: This description doesn't make sense here… this is describing standard claims that are part of the open id spec, not users you create.

File: install_config/configuring_authentication.adoc [hunk 3]
```diff
@@ -1665,8 +1665,8 @@ with a non-empty value is used. The
...
+`preferred_username`:: The preferred user name when provisioning a user if the `sub` value is not present.
```
Comment (liggitt) [inline install_config/configuring_authentication.adoc]: Revert this change, this is a description of the claims, not of precedence relative to other claims. The configuration they specify determines the fallback order

**Comment by bfallonf** [comment]
Thanks @liggitt . I've updated the PR. Please let me know if you have more thoughts.

@openshift/team-documentation PTAL

**Comment by liggitt** [comment]
LGTM

File: install_config/configuring_authentication.adoc [hunk 4]
```diff
@@ -1657,28 +1657,33 @@ Claims are read from the JWT `id_token` returned from the OpenID identity
...
 You can also indicate which claims to use as the user's preferred user name,
 display name, and email address. If multiple claims are specified, the first one
-with a non-empty value is used. The
-link:http://openid.net/specs/openid-connect-core-1_0.html#StandardClaims[standard claims] are:
+with a non-empty value is used. The standard claims are:
 
 [horizontal]
-`sub`:: The user identity.
-`preferred_username`:: The preferred user name when provisioning a user.
+`sub`:: Short for "subject identifier". The remote identity for the user at the
```
Comment (kalexand-rh) [inline install_config/configuring_authentication.adoc]: s/identifier"./identifier."

File: install_config/configuring_authentication.adoc [hunk 5]
```diff
@@ -1657,28 +1657,33 @@ Claims are read from the JWT `id_token` returned from the OpenID identity
...
+issuer.
+`preferred_username`:: The preferred user name when provisioning a user. A
+shorthand name that the user wishes to be referred, such as `janedoe`. Typically
```
Comment (kalexand-rh) [inline install_config/configuring_authentication.adoc]: s/wishes to be referred, /wants to be referred to
(IBM Style says "no" to wish.)

File: install_config/configuring_authentication.adoc [hunk 6]
```diff
@@ -1657,28 +1657,33 @@ Claims are read from the JWT `id_token` returned from the OpenID identity
...
+a value that corresponding to the users login or username in the authentication
```
Comment (kalexand-rh) [inline install_config/configuring_authentication.adoc]: s/corresponding to the users login or username/corresponds to the user's login or user name

File: install_config/configuring_authentication.adoc [hunk 7]
```diff
@@ -1657,28 +1657,33 @@ Claims are read from the JWT `id_token` returned from the OpenID identity
...
+system (username, email, etc).
```
Comment (kalexand-rh) [inline install_config/configuring_authentication.adoc]: s/system (username, email, etc)./system, such as user name or email address.
(IBM Style says "no" to both parentheses and etc.)

File: install_config/configuring_authentication.adoc [hunk 8]
```diff
@@ -1720,13 +1725,13 @@ must be allowed to redirect to `<master>/oauth2callback/<identityProviderName>`.
 <6> The client secret. This value may also be provided in an
 xref:../install_config/master_node_configuration.adoc#master-node-configuration-passwords-and-other-data[environment
 variable, external file, or encrypted file].
-<7> Use the value of the `sub` claim in the returned `id_token` as the user's
-identity.
+<7> List of claims to use as the identity. First non-empty claim is used. At
```
Comment (kalexand-rh) [inline install_config/configuring_authentication.adoc]: s/First/The first

File: install_config/configuring_authentication.adoc [hunk 9]
```diff
@@ -1720,13 +1725,13 @@ must be allowed to redirect to `<master>/oauth2callback/<identityProviderName>`.
...
+least one claim is required. If none of the listed claims have a value,
+authentication fails. For example, this uses the value of the `sub` claim in the returned `id_token` as the user's identity.
```
Comment (kalexand-rh) [inline install_config/configuring_authentication.adoc]: s/this/this example

File: install_config/configuring_authentication.adoc (hunk 1 above)
Comment (kalexand-rh) [reply install_config/configuring_authentication.adoc]: I like the new text, but I might move the annotation marker down a line so it's on the (one item) list itself.

**Review by kalexand-rh — COMMENTED** [review]
I have a couple of style issues, as always, but this looks good. :)

**Comment by bfallonf** [comment]
Thanks @kalexand-rh . I'll merge.
